src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,117 @@
 package core.basesyntax;
 
-public class ArrayList<T> implements List<T> {
+import java.util.NoSuchElementException;
 
-    @Override
-    public void add(T value) {
+public class ArrayList<E> implements List<E> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private E[] objects;
+    private int size;
 
+    public ArrayList() {
+        objects = (E[]) new Object[DEFAULT_CAPACITY];
     }
 
-    @Override
-    public void add(T value, int index) {
+    public ArrayList(int capacity) {
+        objects = (E[]) new Object[capacity];
+    }
 
+    @Override
+    public void add(E value) {
+        add(value, size);
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void add(E value, int index) {
+        if (size == objects.length) {
+            grow();
+        }
+        if (index > size || index < 0) {
+            throwIndexException();
+        }
+        System.arraycopy(objects, index, objects,index + 1, objects.length - index - 1);
```

Comment:
I am not sure about the last parameter (count of elements to shift)
let's operate `size` and not `objects.length`